src/main/java/bob/Task.java

```diff
@@ -0,0 +1,50 @@
+package bob;
+
+/**
+ * Parent class of all task types.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    protected Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    public String getStatusIcon() {
+        return (this.isDone ? "X" : " ");
+    }
+
+    public void toggleStatus() {
+        this.isDone = !this.isDone;
+    }
```

Comment:
hmm, seems like this method is not used anywhere, perhaps you'll want to remove it?

```suggestion
    
```